client/src/actions/index.ts

```diff
@@ -72,15 +78,17 @@ async function configFetch(dispatch: AppDispatch): Promise<Config> {
 /**
  * Fetch dataset metadata and dispatch save to store, including portal URL returned in /config.
  * @param dispatch Function facilitating update of store.
+ * @param oldPrefix API prefix with dataset path that dataset metadata lives on. (Not S3 URI)
  * @param config Response from config endpoint containing collection ID for the current dataset.
  */
-async function datasetMetadataFetch(
+async function datasetMetadataFetchAndLoad(
   dispatch: AppDispatch,
+  oldPrefix: string,
```

Comment:
We're passing the old prefix (data root path) now, as this API call occurs later down the line and by this point the global has been overwritten with the S3URI prefix. (metadata lives on the data root)

Reply:
Since this var name assumes prior knowledge of what "old" means, maybe it can be named more explicitly for future engineers? 

E.g., in the BE code, it's being called "[dataset] Explorer URL path".  

(Though BE should probably change "[dataset] Explorer URL path" to "... URI ...", to be correct, as it does not include an HTTP scheme in the path.)

Reply:
> by this point the global has been overwritten with the S3URI prefix

What if we just main two prefixes on `API`, one for the "public" dataset identifier (i.e. the "explorer url") and one for the "internal" dataset _artifact_ identifier? It's conceivable we'll add more API calls in the future that need the former.  

I think we should also settle on some clear variable names for:
* the "public" dataset identifier (i.e. the "explorer url"): `datasetPublicPath` or `datasetPublicIdentifier`
* the "internal" dataset _artifact_ identifier: e.g. `datasetArtifactUri`

And then use these throughout the BE and FE code:
* `s3URIFetch()` -> `datasetArtifactUriFetch()`
* `updateAPIWithS3()` -> `updateAPIWithDatasetArtifactUri()`

Reply:
Maybe a quick meeting today to align on naming? If/when naming changes are decided upon, we can address changes as a new refactoring story.

Reply:
No strong preference on naming. For some context, the word `old` has a runtime connotation. Old as in before it was rewritten ms ago.

We could do something a bit more literal and how it's written in my comments, such as:
 * s3BaseAPIPrefix - but once this is used as the actual APIPrefix I think the default API naming is proper
 * dataRootAPIPrefix

Reply:
Thank you for weighing in on the naming! I'm going to add a new story to address naming changes on both FE and BE at same time, to keep them aligned. Will include current naming suggestions, and will seek broader team input as well.

Reply:
https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-explorer/196